clippy_lints/src/unnecessary_box_returns.rs

```diff
@@ -0,0 +1,81 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, ty::implements_trait};
+use rustc_errors::Applicability;
+use rustc_hir::{intravisit::FnKind, Body, FnDecl, FnRetTy, HirId};
+use rustc_hir_analysis::hir_ty_to_ty;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Checks for a return type containing a `Box<T>` where `T` implements `Sized`
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It's better to just return `T` in these cases. The caller may not need
+    /// the value to be boxed, and it's expensive to free the memory once the
+    /// `Box<T>` been dropped.
+    ///
+    /// ### Example
+    /// ```rust
+    /// fn foo() -> Box<String> {
+    ///     Box::new(String::from("Hello, world!"))
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo() -> String {
+    ///     String::from("Hello, world!")
+    /// }
+    /// ```
+    #[clippy::version = "1.64.0"]
+    pub UNNECESSARY_BOX_RETURNS,
+    nursery,
```

Comment:
I believe we can directly put it into a proper lint group. We might want to ask lintcheck before, to see if it's too noisy for warn-by-default.

Reply:
I tried setting the category to "perf" and ran `cargo lintcheck` and didn't see anything for this lint, so you might be right. Does "perf" seem like the right category here? Personally, I think before that happens, we should at least look into making this lint dependent on the size of the struct.

Reply:
`cargo lintcheck` also has a `--filter <lint>` option to ensure all cases are emitted.

For the `perf` group a size check should be required. Even than, I'd be careful to move it directly from `Nursery` to `perf`. I think `pedantic` would be the correct group :)

Reply:
It's now in pedantic